cpp/benchmarks/iterator/iterator_benchmark.cu

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmark/benchmark.h>
+
+#include <random>
+#include <tests/utilities/column_wrapper.hpp>
+
+#include "../fixture/benchmark_fixture.hpp"
+#include "../synchronization/synchronization.hpp"
+
+#include <cudf/detail/iterator.cuh> // include iterator header
+// for reduction tests
+#include <cub/device/device_reduce.cuh>
+#include <cudf/detail/utilities/device_operators.cuh>
+#include <thrust/device_vector.h>
+
```

Comment:
Do these benchmarks test the pair_iterator at all? If not, why are they in this PR? (confused)

Reply:
I added pair_iterator benchmarks as well.